src/Lime.Protocol/Client/EstablishedClientChannelBuilder.cs

```diff
@@ -102,7 +102,11 @@ public IEstablishedClientChannelBuilder WithCompression(SessionCompression compr
         /// <exception cref="System.ArgumentNullException"></exception>
         public IEstablishedClientChannelBuilder WithCompression(Func<SessionCompression[], SessionCompression> compressionSelector)
         {
-            if (compressionSelector == null) throw new ArgumentNullException(nameof(compressionSelector));
+            if (compressionSelector == null)
```

Comment:
What do you think about using Dawn.Guard in here (and in all other cases it applies)?

Reply:
I agree. I like the readability of `Dawn` and how clean it is, plus I like fluent code.

However the other implementations were using conditionals, therefore I simply followed the existing samples.

By using `Guard.Argument(variable, nameof(variable)).NotNull()` I can achieve a similar enough exception message to the `ArgumentNullException`, I'll change it and, if requested, I'll revert my changes